README.md

```diff
@@ -204,103 +207,57 @@ Assume we have space with `id = 512` and following format on the server:
 `CREATE TABLE t(id INT PRIMARY KEY, a TEXT, b DOUBLE);`  
 Preparing analogue of `t:replace(1, "111", 1.01);` request can be done this way:
 
-```
+```c++
 std::tuple data = std::make_tuple(1 /* field 1*/, "111" /* field 2*/, 1.01 /* field 3*/);
 rid_t replace = conn.space[512].replace(data);
 ```
 To execute select query `t.index[1]:select({1}, {limit = 1})`:
 
-```
+```c++
 auto i = conn.space[512].index[1];
 rid_t select = i.select(std::make_tuple(1), 1, 0 /*offset*/, IteratorType::EQ);
 ```
 
 ### Data readers
 
-Responses from server contain raw data (i.e. encoded into msgpuck tuples). To
-decode client's data, users have to write their own decoders (based on featured
-schema). Let's define structure describing data stored in space `t`:
+Responses from server contain raw data (i.e. encoded into msgpuck tuples).
+Let's define structure describing data stored in space `t`:
 
-```
+```c++
 struct UserTuple {
     uint64_t field1;
     std::string field2;
     double field3;
-};
-```
 
-Prototype of the base reader is given in `src/mpp/Dec.hpp`:
-```
-template <class BUFFER, Type TYPE>
-struct SimpleReaderBase : DefaultErrorHandler {
-    using BufferIterator_t = typename BUFFER::iterator;
-    /* Allowed type of values to be parsed. */
-    static constexpr Type VALID_TYPES = TYPE;
-    BufferIterator_t* StoreEndIterator() { return nullptr; }
+	static constexpr auto mpp = std::make_tuple(
+		&UserTuple::field1, &UserTuple::field2, &UserTuple::field3);
```

Comment:
C++ short-circuits my mind sometimes. [Pointers to data members](https://en.cppreference.com/w/cpp/language/pointer#Pointers_to_data_members) is a really neat feature.

Reply:
I left a link here so that people could learn them if they've never seen them before.